SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs

```diff
@@ -261,34 +261,38 @@ private AppCenter()
         internal IApplicationSettings ApplicationSettings => _applicationSettings;
         internal INetworkStateAdapter NetworkStateAdapter => _networkStateAdapter;
 
-        private bool InstanceEnabled
+        private bool InstanceEnabled => _applicationSettings.GetValue(EnabledKey, true);
+
+        private Task SetInstanceEnabled(bool value)
         {
-            get
+            var enabledTerm = value ? "enabled" : "disabled";
+            if (InstanceEnabled == value)
             {
-                return _applicationSettings.GetValue(EnabledKey, true);
+                AppCenterLog.Info(AppCenterLog.LogTag, $"App Center has already been {enabledTerm}.");
+                return Task.FromResult(default(object));
             }
-            set
-            {
-                var enabledTerm = value ? "enabled" : "disabled";
-                if (InstanceEnabled == value)
-                {
-                    AppCenterLog.Info(AppCenterLog.LogTag, $"App Center has already been {enabledTerm}.");
-                    return;
-                }
 
-                // Update channels state.
-                _channelGroup?.SetEnabled(value);
+            // Update channels state.
+            _channelGroup?.SetEnabled(value);
 
-                // Store state in the application settings.
-                _applicationSettings.SetValue(EnabledKey, value);
+            // Store state in the application settings.
+            _applicationSettings.SetValue(EnabledKey, value);
 
-                // Apply change to services.
-                foreach (var service in _services)
-                {
-                    service.InstanceEnabled = value;
-                }
-                AppCenterLog.Info(AppCenterLog.LogTag, $"App Center has been {enabledTerm}.");
+            // Apply change to services.
+            foreach (var service in _services)
+            {
+                service.InstanceEnabled = value;
             }
+            AppCenterLog.Info(AppCenterLog.LogTag, $"App Center has been {enabledTerm}.");
+
+            // Send started services.
+            if (_startedServiceNames != null && value)
+            {
+                var startServiceLog = new StartServiceLog { Services = _startedServiceNames };
+                _startedServiceNames = null;
+                return _channel.EnqueueAsync(startServiceLog);
```

Comment:
@guperrot not from result here

Reply:
Hum, better add a comment then, it's weird to have a non async method return a task